toolkit/tools/pkg/imagecustomizerlib/artifactsinputoutput.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/imagegen/diskutils"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safeloopback"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safemount"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	ShimDir        = "EFI/BOOT"
+	SystemdBootDir = "EFI/systemd"
+	UkiPattern     = `vmlinuz-.*\.efi$`
+)
+
+func outputArtifacts(items []imagecustomizerapi.Item, outputDir string, buildDir string, buildImage string) error {
+	logger.Log.Infof("Outputting artifacts")
+
+	loopback, err := safeloopback.NewLoopback(buildImage)
+	if err != nil {
+		return fmt.Errorf("failed to connect to image file to output artifacts:\n%w", err)
+	}
+	defer loopback.Close()
+
+	diskPartitions, err := diskutils.GetDiskPartitions(loopback.DevicePath())
+	if err != nil {
+		return err
+	}
+
+	systemBootPartition, err := findSystemBootPartition(diskPartitions)
+	if err != nil {
+		return err
+	}
+
+	systemBootPartitionTmpDir := filepath.Join(buildDir, tmpEspPartitionDirName)
+	systemBootPartitionMount, err := safemount.NewMount(systemBootPartition.Path, systemBootPartitionTmpDir, systemBootPartition.FileSystemType, unix.MS_RDONLY, "", true)
+	if err != nil {
+		return fmt.Errorf("failed to mount esp partition (%s):\n%w", systemBootPartition.Path, err)
+	}
+	defer systemBootPartitionMount.Close()
+
+	// Detect system architecture
+	arch := runtime.GOARCH
+	var shimBinaryName, systemdBootBinaryName string
+
+	switch arch {
+	case "amd64", "x86_64":
+		shimBinaryName = "bootx64.efi"
```

Comment:
It would be good to share this information with `liveosisobuilder.go`.

Reply:
`liveosisobuilder.go` already has logic that handles the name of the shim binary. See, `bootx64Binary`, `bootAA64Binary`, and `bootloaderFilesConfig`. You should re-use these variables/logic instead of reimplementing it.